src/xdg.rs

```diff
@@ -40,11 +38,12 @@ impl DesktopEntries {
         // Initialize icon loader.
         let loader = IconLoader::new(&dirs, "hicolor");
 
-        let mut desktop_entries = DesktopEntries { scale_factor, loader, entries: Vec::new() };
-        let icon_size = desktop_entries.icon_size();
-
         // Create placeholder icon.
-        let placeholder_icon = Rc::new(Icon::new_placeholder(icon_size)?);
+        const PLACEHOLDER_SVG: &[u8] = include_bytes!("../svgs/placeholder.svg");
+        let placeholder = Svg::parse(PLACEHOLDER_SVG)?;
+
+        let scale_factor = 0.;
```

Comment:
Defaulting to 0 seems like a bad idea?

Reply:
I agree, but I’m not sure what better way to carry a “not yet initialised state”, which ideally would also go into most other structs so that we can preload them before we even got the Wayland connection (for instance on a different thread).  That was also the solution I went for with the `Svg` struct when it hasn’t been rendered yet, but I also don’t like it too much.